rskj-core/src/main/java/co/rsk/net/handler/TxsPerAccount.java

```diff
@@ -54,10 +56,12 @@ public void setTransactions(List<Transaction> txs){
 
         List<Transaction> ret = new LinkedList<>();
         for (Transaction tx : txs) {
-            BigInteger nonce = new BigInteger(1, tx.getNonce());
-            if (nextNonce.compareTo(nonce) == 0) {
-                nextNonce = nonce.add(BigInteger.valueOf(1));
-                ret.add(tx);
+            for(byte[] txNonce : tx.getNonces()) {
+                BigInteger nonce = new BigInteger(1, txNonce);
```

Comment:
Same here. I don't think this class should handle multiple nonces. Only the nonces that are related to this account